main_test.go

```diff
@@ -0,0 +1,83 @@
+package main
+
+import (
+	"github.com/oyugirachel/deck"
+	"testing"
+	
+)
+
+var (
+	ace   = deck.Card{Suit: deck.Spade, Rank: deck.Ace}
```

Comment:
All of the cards you setup here are in the same Suit. This means that your tests wouldn't catch issue #1 - where the game rules say we match on rank, so cards of the same rank but different suits still count for a snap.

Think about adding a sub-test or tests to cover issue 1